File: drift-static-profiler/tests/fixtures/scala-play/app/repos/OrdersRepository.scala
package repos

import scala.concurrent.Future
import slick.jdbc.PostgresProfile.api._

case class Order(id: Long, customerEmail: String, totalCents: Long, currency: String)

class OrdersRepository(db: Database) {

  def save(order: Order): Future[Long] = {
    val q = sqlu"""INSERT INTO orders (customer_email, total_cents, currency)
                   VALUES (${order.customerEmail}, ${order.totalCents}, ${order.currency})"""
    db.run(q).map(_.toLong)(scala.concurrent.ExecutionContext.global)
  }

  def findById(id: Long): Future[Option[Order]] = {
    val q = sql"""SELECT id, customer_email, total_cents, currency
                  FROM orders WHERE id = $id""".as[(Long, String, Long, String)]
    db.run(q.headOption).map(_.map(t => Order(t._1, t._2, t._3, t._4)))(scala.concurrent.ExecutionContext.global)
  }
}

File: drift-static-profiler/tests/fixtures/kotlin-ktor/src/main/kotlin/com/example/repos/OrdersRepository.kt
package com.example.repos

import java.sql.Connection
import java.sql.Statement

data class Order(
    val id: Long,
    val customerEmail: String,
    val totalCents: Long,
    val currency: String,
)

class OrdersRepository(private val conn: Connection) {

    fun save(order: Order): Long {
        val stmt = conn.prepareStatement(
            "INSERT INTO orders (customer_email, total_cents, currency) " +
                "VALUES (?, ?, ?)",
            Statement.RETURN_GENERATED_KEYS,
        )
        stmt.setString(1, order.customerEmail)
        stmt.setLong(2, order.totalCents)
        stmt.setString(3, order.currency)
        stmt.executeUpdate()
        val keys = stmt.generatedKeys
        keys.next()
        return keys.getLong(1)
    }

    fun findById(id: Long): Order? {
        val stmt = conn.prepareStatement(
            "SELECT id, customer_email, total_cents, currency FROM orders WHERE id = ?"
        )
        stmt.setLong(1, id)
        val rs = stmt.executeQuery()
        if (!rs.next()) return null
        return Order(
            id = rs.getLong(1),
            customerEmail = rs.getString(2),
            totalCents = rs.getLong(3),
            currency = rs.getString(4),
        )
    }
}
